app/src/main/java/org/schabi/newpipe/util/ExtractorHelper.java

```diff
@@ -131,6 +136,22 @@ private static void checkServiceId(int serviceId) {
                 ChannelInfo.getMoreItems(NewPipe.getService(serviceId), url, nextStreamsUrl));
     }
 
+    public static Single<ListInfo<StreamInfoItem>> getFeedInfoFallbackToChannelInfo(final int serviceId,
+                                                                                    final String url) {
+        final Maybe<ListInfo<StreamInfoItem>> maybeFeedInfo = Maybe.fromCallable(() -> {
+            final StreamingService service = NewPipe.getService(serviceId);
+            final FeedExtractor feedExtractor = service.getFeedExtractor(url);
+
+            if (feedExtractor == null) {
+                return null;
+            }
+
+            return FeedInfo.getInfo(feedExtractor);
```

Comment:
Is it ok for exceptions in this block to cause errors? Shouldn't this be put in a try-catch that just returns null on error?

Reply:
Yes, it should work just like the others, check their implementation and you will see the usage of `fromCallable` just like this (well with some caching code involved, but those are just for checking if it is cached or not).

Also checkout the `switchIfEmpty` below, pretty much like the switching done by the caching code.